provision-infra-web.yml

```diff
@@ -1398,11 +1401,102 @@
     - dashboards
     - grafana
 
+- name: Shortener Databases
+  hosts: databases.infra.netsoc.co
+  become: yes
+  tasks:
+    - name: Shortener - Ensure database
+      postgresql_db:
+        name: "{{ shortener.postgres.db }}"
+        state: present
+        login_host: "{{ databases.infra.postgres }}"
+        login_password: "{{ postgres.infra.postgres_password }}"
+
+    - name: Shortener - Ensure user
+      postgresql_user:
+        name: "{{ shortener.postgres.user }}"
+        password: "{{ shortener.postgres.pass }}"
+        login_host: "{{ databases.infra.postgres }}"
+        login_password: "{{ postgres.infra.postgres_password }}"
+      no_log: true
+
+    - name: Shortener - Ensure user has access to database
+      postgresql_privs:
+        db: "{{ shortener.postgres.db }}"
+        role: "{{ shortener.postgres.user }}"
+        objs: ALL_IN_SCHEMA
+        privs: SELECT,INSERT,UPDATE,DELETE
+        login_host: "{{ databases.infra.postgres }}"
+        login_password: "{{ postgres.infra.postgres_password }}"
+  vars_files:
+    - "vars/network.yml"
+    - "vars/secrets_mapping.yml"
+    - "vars/secrets.yml"
+  tags:
+    - provision
+    - shortener
+
+- name: "Shortener"
+  hosts: web.infra.netsoc.co
+  become: yes
+  tasks:
+    - name: Shortener - Remember Compose definition
+      set_fact:
+        shortener_definition:
+          version: "3.8"
+          networks:
+            traefik:
+              external: true
+          services:
+            instance:
+              image: uccnetsoc/shortener:latest
+              environment:
+                DB_HOST: "{{ interfaces.infra.databases.net0.addresses[0] | ipaddr('address') }}"
+                DB_PORT: "5432"
+                DB_USER: "{{ shortener.postgres.user }}"
+                DB_PASS: "{{ shortener.postgres.pass }}"
+                DB_NAME: "{{ shortener.postgres.db }}"
+              networks:
+                traefik:
+              labels:
+                traefik.enable: "true"
+                traefik.http.routers.shortener.rule: "Host(`{{ shortener.traefik.domain }}`)"
```

Comment:
Do we need the domain as a secret?

Reply:
do you want me to keep it as a variable, or just hardcode it in?

Reply:
Hardcoded is grand :p

Reply:
its used in the bot's environment variables too, so i thought it might be appropriate

Reply:
I'd say just keep it inline, makes it more readable :)